vowpalwabbit/io_to_queue.h

```diff
@@ -69,22 +52,35 @@ inline bool read_input_file_ascii(vw& all, char *&line) {
 }
  
 inline bool read_input_file_binary(vw& all, char *&line) {
- 
- size_t total_num_input_bytes = 0;
- 
- //read_all_data returns the number of bytes successfully read from the input files
- total_num_input_bytes = all.example_parser->input->read_all_data(line, total_num_input_bytes);
- 
- bool finish = true;
- 
+
+  size_t example_length_bytes;
+  io_buf* input = all.example_parser->input.get();
+
+  // Return true if no examples found in file.
+  if((*input).buf_read(line, sizeof(example_length_bytes)) < sizeof(example_length_bytes)) {
+    // The end of io_lines should contain a null pointer.
+    std::vector<char> *byte_array = new std::vector<char>();
+    all.example_parser->io_lines.push(std::move(byte_array));
```

Comment:
nit: the move here is unnecessary. You're attempting to move a pointer, which is just a copy